[Guides] removed routing_filter gem reference as it is outdated [ci-skip] (#54601)
rails/rails: guides/source/i18n.md

Do take special care about the **order of your routes**, so this route declaration does not "eat" other ones. (You may want to add it directly before the `root :to` declaration.)

> NOTE: Have a look at the gem that simplifies working with routes:  [route_translator](https://github.com/enriclluelles/route_translator).

Reviewer: ```suggestion
NOTE: Have a look at a gem which simplifies working with routes:  [route_translator](https://github.com/enriclluelles/route_translator).
```

Author: @skipkayhil made the suggestions. Thank you.
